Jest Refactor

DominicMonares opened the pull request:
This PR completely replaces mocha/ chai with jest: 
- All old test files have had their contents moved into one jest test file, then old files were deleted.
- Mocha and chai have been uninstalled, jest has been installed
- The start script has been updated accordingly
- testServer has been split into two files: testServer and testIndex
  - This was done because jest doesn't like it when you import server files that call app.listen
    - I haven't done extensive testing here, it's likely some changes will need to be made here
- Side note: I removed the comment about saving config files as .env, since we will be using .js for our configs

Please let me know what y'all think and if there are any revisions to be made.

RobStafford reviewed (changes requested):
Hi Dominic--

The only suggestion I'd have here is that a SIER told me that "react-dev": "webpack --watch", (package.json) is a better script than the one we had, and I'd recommend keeping it rather than writing over it.

dwightdoran commented on tests/testServer.js:
```diff
@@ -1,6 +1,6 @@
-const app = require('./testApp.js')(app);
+const app = require('./testApp.js');
```
Not sure if you changed the file name from ./testApp.js to ./testIndex.js. If not, disregard this comment as it's showing two different things on my end.

dwightdoran commented:
Looks awesome @DominicMonares!! I just have that one comment. Other than that, it looks great.

DominicMonares commented:
Thanks for the reviews @RobStafford and @dwightdoran ! Great catches. I don't remember why I changed the webpack script to be completely honest, I reverted that back to webpack --watch. I did change the testApp to testIndex file so I missed changing that route in testServer, route has been corrected.
